contrib/udfs/src/test/java/org/apache/drill/exec/udfs/TestDateUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.udfs;
+
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestDateUtils {
+
+  @Test
+  public void testDateFromString() {
+    LocalDate testDate = LocalDate.of(2022, 3,14);
+    LocalDate badDate = LocalDate.of(1970, 1, 1);
+    assertEquals(testDate, DateUtilFunctions.getDateFromString("2022-03-14"));
+    assertEquals(testDate, DateUtilFunctions.getDateFromString("3/14/2022"));
+    assertEquals(testDate, DateUtilFunctions.getDateFromString("14/03/2022", true));
+    assertEquals(testDate, DateUtilFunctions.getDateFromString("2022/3/14"));
+
+    // Test bad dates
+    assertEquals(badDate, DateUtilFunctions.getDateFromString(null));
+    assertEquals(badDate, DateUtilFunctions.getDateFromString("1975-13-56"));
+    assertEquals(badDate, DateUtilFunctions.getDateFromString("1975-1s"));
```

Comment:
My thinking with that is that `null` tends to break secondary functions.  Returning a default value would prevent that.  This is a debate that Paul Rogers and I have had many times.  Is it better to return null, a default or error out.

Reply:
CC @paul-rogers in case he's also got any thoughts to add to this thread.

Reply:
Footnote, since this point comes up quite often. Those of us who've done data analytics with Pandas _did_ get used to floating point NaN being used as a "sentinel" for missing or invalid data but we should recognise this for what it was: a performance hack that entered Pandas from its Numpy foundation. It resulted in pain: automatic casting of ints to floats even though precision loss could happen because hardware and C integer types have no NaN value, special code to implement `skipna` for making NaNs behave like nulls rather according to the IEEE float rules, etc.

And indeed, [here is Wes McKinney talking about moving away from that approach](https://wesmckinney.com/blog/bitmaps-vs-sentinel-values/). Here's a relevant excerpt for us, one hop over in the SQL world where null is a standardised first class citizen:

> From the perspective of databases and data warehousing, reserving certain values to mark a null (or NA) is widely considered unacceptable. NaN is valid data, as is INT32_MIN and other common values used as sentinels.

Reply:
@jnturton I'm a little stuck here.  I did a bunch of experimentation, and right now, if the input is `null`, the function will return `null`.   However, if the input is unparsable, then the output is Jan 1, 1970.   My goal is to make it so that if the input is not parsable, it will also return `null`, however I can't seem to get it to do that.

I've tried using a `NullableTimestampHolder` as output and simply setting the value to `null` or not setting it at all. but When I do that, the function does not work at all.   Meaning that in the unit tests, ALL results come back as `null`.  

@vvysotskyi Any suggestions?

Reply:
Fixed